cube/src/main/java/zone/cogni/asquare/cube/convertor/data2shacl/ShaclGenerator.java

```diff
@@ -325,17 +328,35 @@ private void setShaclDatatype(@Nonnull RdfStoreService rdfStore,
                                 @Nonnull Resource path,
                                 @Nonnull Resource propertyShape) {
     List<String> datatypes = selectUris(rdfStore, "select-datatype.sparql.spel", getTypeAndPropertyParameters(targetClass, path));
-    if (datatypes.size() != 1) {
-      log.warn(getMessage("type '{}' and property '{}' does not have exactly one datatype: {}",
-                          shortenUri(shacl, targetClass), shortenUri(shacl, path), shortenUri(shacl, datatypes)));
+    if (datatypes.isEmpty()) {
+      log.warn(getMessage("type '{}' and property '{}' does not have exactly a datatype",
+                          shortenUri(shacl, targetClass), shortenUri(shacl, path)));
       return;
     }
 
-    Resource datatypeValue = ResourceFactory.createResource(datatypes.get(0));
-    shacl.add(propertyShape, Shacl.datatype, datatypeValue);
+    if (datatypes.size() > 1) {
+      datatypes.forEach(datatype -> {
+        Resource datatypeValue = ResourceFactory.createResource(datatype);
+
+        String orInstanceName = datatypeValue.getLocalName().toLowerCase();
```

Comment:
This will be an issue in the improbable case of having 2 datatypes with same localName but different namespace